### PR TITLE
Get the coveralls working again (really, really)

### PR DESCRIPTION
**Pull request by kratsg**
# Description

#374 didn't do what it claimed to do. Hopefully this one is more honest.

# Checklist Before Requesting Reviewer

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request
- [x] Selected an Assignee for the PR to be responsible for the log summary

# Before Merging

For the PR Assignees:

- [x] Summarize commit messages into a comprehensive review of the PR
```
Move the after_success back to the matrix, and skip after_success on all non-matrix jobs.
```

### CODE REVIEW

**Comment by matthewfeickert** [comment]
@kratsg For those of us who learned Travis by `--force-with-lease` trial and error can you summarize why moving coveralls to the matrix stage fixes things? I thought the goal was to have it run only after Python 3.6, but this has it run on Python 2.7 as well. Sorry for missing what is happening.

**Comment by kratsg** [comment]
> @kratsg For those of us who learned Travis by `--force-with-lease` trial and error can you summarize why moving coveralls to the matrix stage fixes things? I thought the goal was to have it run only after Python 3.6, but this has it run on Python 2.7 as well. Sorry for missing what is happening.

Basically, it was running in the 3.6 documentation only, and not anything else. So I moved it back to where it was before, and added `after_success: skip` overrides everywhere else.
